WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/usecases/GetRestoreStatusUseCase.kt

```diff
@@ -0,0 +1,64 @@
+package org.wordpress.android.ui.jetpack.restore.usecases
+
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.flow
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.model.activity.RewindStatusModel.Rewind.Status.FAILED
+import org.wordpress.android.fluxc.model.activity.RewindStatusModel.Rewind.Status.FINISHED
+import org.wordpress.android.fluxc.model.activity.RewindStatusModel.Rewind.Status.RUNNING
+import org.wordpress.android.fluxc.store.ActivityLogStore
+import org.wordpress.android.fluxc.store.ActivityLogStore.FetchRewindStatePayload
+import org.wordpress.android.ui.jetpack.restore.RestoreRequestState.Complete
+import org.wordpress.android.ui.jetpack.restore.RestoreRequestState.Failure.NetworkUnavailable
+import org.wordpress.android.ui.jetpack.restore.RestoreRequestState.Failure.RemoteRequestFailure
+import org.wordpress.android.ui.jetpack.restore.RestoreRequestState.Progress
+import org.wordpress.android.util.NetworkUtilsWrapper
+import javax.inject.Inject
+
+const val DELAY_MILLIS = 10000L
+
+class GetRestoreStatusUseCase @Inject constructor(
```

Comment:
Warning (⚠️): Consider adding tests for this use case.

PS: Also the same applies to the `GetBackupDownloadStatusUseCase` use case, which doesn't have any as well.

Reply:
Noted and added the task. 👍

Reply:
Thanks @zwarm ! Just to make it clear, this will be done on another PR, right?

Reply:
Yes, use case unit tests will be in a separate PR - not yet started